### ISSUE TITLE
module-sybase: add support for TDS protocol 7.3+ for date/time values with freetds

### ISSUE THREAD

**Issue by davidnich** [qorelanguage/qore#2019]

**Comment by pvanek** [comment]
well, with 7.2 version we have this:

Just simple insert and select datetime colum with value 2016-02-11T09:26:14.502134

```
Assertion "checking data in column timestamp_f" failure at /var/lib/jenkins/workspace/qore/qlib/QUnit.qm:1072
-----
	>> Expected: 2016-02-11 09:26:14.502134 Thu +01:00 (CET)
	>> Actual: 2016-02-11 09:26:14.503000 Thu +01:00 (CET)
```

Something like "rounding"

**Comment by pvanek** [comment]
ad my previous note - it's not related to the bug.

Cite:

> SQL Server only stores time to approximately 1/300th of a second. These always fall on the 0, 3 and 7 milliseconds. If you need that millisecond accuracy, there's no pleasant way around it. The best options I've seen are to store the value in custom number fields and rebuild it every time you fetch the value, or to store it as a string of a known format. You can then (optionally) store an 'approximate' date in the native date type for the sake of speed, but it introduces a conceptual complexity that often isn't wanted.

> SQL Server 2008 has much more precision available. The datetime2 type will accurately store values like this: 2008-12-19 09:31:38.5670514 (accuracy to 100 nanoseconds).

**Comment by davidnich** [comment]
fixed with #3951
